### ISSUE TITLE
Updated Nadekobot using the updater and it now crashes on startup

### ISSUE THREAD

**Issue by lori9** [Kwoth/NadekoBot#2407]
Title is self explanatory, it was working just fine until I updated just now. I tried doing FLUSHALL with redist and restarting redist, reinstalling, etc. Nothing worked. This bot helps me a lot when it comes to server management, could anyone assist?

Here's what I got on startup, consistently.
-1 12:16:22 RedisImagesCache | System.Net.Http.HttpRequestException: Response status code does not indicate success: 503 (Service Unavailable).
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at System.Net.Http.HttpClient.<GetByteArrayAsyncCore>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NadekoBot.Core.Services.Impl.RedisImagesCache.<>c__DisplayClass51_0.<<Reload>b__2>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NadekoBot.Core.Services.Impl.RedisImagesCache.<Reload>d__51.MoveNext()

Unhandled Exception: System.Net.Http.HttpRequestException: Response status code does not indicate success: 503 (Service Unavailable).
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at System.Net.Http.HttpClient.<GetByteArrayAsyncCore>d__30.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NadekoBot.Core.Services.Impl.RedisImagesCache.<>c__DisplayClass51_0.<<Reload>b__2>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NadekoBot.Core.Services.Impl.RedisImagesCache.<Reload>d__51.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NadekoBot.Core.Services.ShardsCoordinator..ctor()
   at NadekoBot.Program.Main(String[] args)
   at NadekoBot.Program.<Main>(String[] args)

**Comment by shivaco** [comment]
Also, if anyone got Nadeko to start, the console will stay like this:
![Screenshot](https://i.imgur.com/GfnCd8o.png)

**Comment by lori9** [comment]
Yeah, after restarting Redis I sometimes get just that first line. But it hangs there until I get a 502. I'm guessing I am not the only one impacted?

**Comment by shivaco** [comment]
Looks like.

**Comment by Gremagol** [comment]
I've had the same Issue. Try checking your `images.json` file.

**Comment by lori9** [comment]
I did, but that didn't fix my problem.

**Comment by Kuzu-no-ha** [comment]
I have been having this same issue, even after a full reinstall.

**Comment by shikhir-arora** [comment]
The Redis log you posted shows that the images which are hosted on DigitalOcean spaces now return a 503 for your connection, so it's probably a server issue on DO's end.

**Comment by shikhir-arora** [comment]
Actually try this:

`sudo service redis stop`

`sudo ps aux | grep redis` and if anything is running, get the process ID and `sudo kill -9 <ID>` - make sure the bot is offline.

Restart `sudo systemctl start redis`

Then `sudo redis-cli FLUSHALL` 

And retry.

**Comment by Kwoth** [comment]
Has nothing to do with redis, did you change some of the images in images.json? They can't be downloaded for whatever reason.

**Comment by shikhir-arora** [comment]
Also try wget or curl on the server to one of the image links. There's a slight chance it could also be IPv6 that is causing the issue, since I'm able to download fine.

**Comment by lori9** [comment]
I changed some of the image files to other imgur links, and I changed some internal files like the slot machine graphics. Should I just try and somehow revert to vanilla graphics?

Worth mentioning I've seen at least ten others today get hit by the same exact thing I'm getting now. It seems to be some sort of general issue with user hosted Nadeko at the moment.

**Comment by DaddyDeku** [comment]
I have the same issue on Windows 10 and its a vanilla install also my first time running it so nothing is edited.

**Comment by lori9** [comment]
I looked in the images.json file, which was using the new default files, and I noticed there were many being hosted from a new domain instead of imgur. I replaced them with imgur equivalents and now the bot functions perfectly.

Looks like this digital oceans host is maybe getting DDoSed by the traffic?

cross posted from the other issue

**Comment by Kwoth** [comment]
![image](https://user-images.githubusercontent.com/2537696/42562292-9076a398-8525-11e8-808b-484388e864eb.png)

**Comment by ibigfire** [comment]
Works now.  Is this what will happen every time that server has issues, even when self hosted?  Or only for the conversion process to the new format?

**Comment by lori9** [comment]
Yeah, is switching the digital ocean links over to imgur equivalents enough? Or will the entire bot become impossible to start each time this happens.

**Comment by xnaas** [comment]
Once the images are actually cached, I don't think it matters if imgur, digitalocean, or whoever is down.

**Comment by lori9** [comment]
Imgur is fairly solid, so I think the fix here would just be switching the new slot machine images to that host instead of the new one honestly.

**Comment by xnaas** [comment]
I get the sentiment, but I don't think it's necessary. Once the images are cached on your bot's machine/server, it doesn't matter who the host is. The only reason this was a problem this time was bad timing; that's it.

**Comment by Macley-Kun** [comment]
@ibigfire i think the issue is because of the following:
A change in the image.json happend, making us use less local files, and more web hosted files.
The images default, are on Kwoth's vps.
When you updated, the vps of Kwoth went down.
Therefore, you couldn't download the pictures needed (slot, etc).

In conclusion, you download the images once, unless you would do `.imagereload`. It's just coicidence that all of the above happend at the same time :D

**Comment by Macley-Kun** [comment]
@shivaco btw is it working for you also? As you also had problems with Nadeko on your machine.

**Comment by Kwoth** [comment]
You guys should stop talking about this, so much misinformation here, most of the things you guys said are not true/

Imgur is not a cdn, it's an image sharing platform. New images are on digitaloceanSPACES which is basically a cdn. They were down due to some techinical issues, we were just unlucky. Saying we shouldn't use this because every time it happens the bot won't be able to be ran is doubly false. First because images will be cached (if you read the patch notes) only first time you run the bot after updating it, after that you will have to do it manually with `.imagesreload` and secondly it would be the same as saying "lets not host nadeko's code on github because every time github is down we won't be able to get it". 

Tl;dr We were unlucky, all is well now.
